wasm/wasm.go

```diff
@@ -233,3 +239,67 @@ func EncodeF64(input float64) uint64 {
 func DecodeF64(input uint64) float64 {
 	return math.Float64frombits(input)
 }
+
+// HostModuleConfig are WebAssembly 1.0 (20191205) exports from the host bound to a module name used by InstantiateHostModule.
+type HostModuleConfig struct {
+	// Name is the module name that these exports can be imported with. Ex. wasi.ModuleSnapshotPreview1
+	Name string
+
+	// Functions adds functions written in Go, which a WebAssembly Module can import.
+	//
+	// The key is the name to export and the value is the func. Ex. WASISnapshotPreview1
+	//
+	// Noting a context exception described later, all parameters or result types must match WebAssembly 1.0 (20191205) value
+	// types. This means uint32, uint64, float32 or float64. Up to one result can be returned.
+	//
+	// Ex. This is a valid host function:
+	//
+	//	addInts := func(x uint32, uint32) uint32 {
+	//		return x + y
+	//	}
+	//
+	// Host functions may also have an initial parameter (param[0]) of type context.Context or wasm.ModuleContext.
+	//
+	// Ex. This uses a Go Context:
+	//
+	//	addInts := func(ctx context.Context, x uint32, uint32) uint32 {
+	//		// add a little extra if we put some in the context!
+	//		return x + y + ctx.Value(extraKey).(uint32)
+	//	}
+	//
+	// The most sophisticated context is wasm.ModuleContext, which allows access to the Go context, but also
+	// allows writing to memory. This is important because there are only numeric types in Wasm. The only way to share other
+	// data is via writing memory and sharing offsets.
+	//
+	// Ex. This reads the parameters from!
+	//
+	//	addInts := func(ctx wasm.ModuleContext, offset uint32) uint32 {
+	//		x, _ := ctx.Memory().ReadUint32Le(offset)
+	//		y, _ := ctx.Memory().ReadUint32Le(offset + 4) // 32 bits == 4 bytes!
+	//		return x + y
+	//	}
+	//
+	// See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#host-functions%E2%91%A2
+	Functions map[string]interface{}
+
+	// TODO
+	Globals map[string]*HostModuleConfigGlobal
+	// TODO
+	Table *HostModuleConfigTable
```

Comment:
still need the name both here and below. Even if there is a limit of only one, the structure is more coherent if it is a map. especially consider this is likely to be encoded as json or yaml.